README.md

```diff
@@ -1,3 +1,360 @@
+   copied from Pickle will have an additional copyright and licence, and credit to the original Pickle contributors.
+
+## npecl itself
+
+ * use `symfony/console` to simplify writing the CLI tool itself
+ * use `composer/composer` to resolve dependencies; it knows how to do all this stuff already, lets not re-invent the
+   wheel - once we have a better idea of what calls we need to make, it may make sense to split some of
+   `composer/composer` out into separate libraries we can consume; although that might be a huge undertaking in itself,
+   so is really just a "strech goal".
+ * There may be multiple versions of php on a system. The version of php invoking `npecl` would be assumed, and its
+   corresponding `phpize` tool would be used.
+ * `npecl` would probably need to be run with `sudo` access (so it can `make install` to root-owned paths). Care should
+   be taken to ensure only the minimal amount is run with elevated privileges, so we don't accidentally give root to a
+   malicious PECL package. Ideally, `npecl` would run without `sudo`, and just prompt for access, if possible.
+
+### CLI commands
+
+#### `install {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Installs the requested package. The package may be requested with or without `ext-` prefix. `npecl install ext-xdebug`
+and `npecl install xdebug` would be equivalent.
+
+If `version-constraint` is given, try to install that version if it matches the allowed versions.
+
+On Windows this downloads the correct DLL (attached as file to the release tag), if available. Otherwise, it will
+download the source and compile the extension.
+
+It will then attempt to create a 20-{extension_name}.ini file with `extension={extension_name}` in the directory
+returned by `php-config –ini-dir`, the name that is used is the one from the "name" element in the metadata file.
+
+If `@dev-branch-name` is given, try to install from the branch called "branch-name", for example use `@dev-master` to
+compile from your master branch. As branches would usually not have binaries, this would not work for Windows.
+
+You can't use both a `version-constraint` and `@dev-branch-name` at the same time.
+
+Each step should run with as few privileges as possible.
+
+#### `build {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as install, but only compiles (or downloads if it's Windows).
+
+#### `changelog {?version}`
+
+Shows the release notes of the version it was going to install, or from the specific version, if given.
+
+#### `download {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as build, but puts the files in a local directory for manual building and installation.
+
+#### `info`
+
+Shows the description from the metadata file
+
+#### `list`
+
+Shows all available commands
+
+#### `self-update`
+
+Attempts to upgrade the npecl tool itself
```

Comment:
I think we should leave this out for now. I expect we can replicate what `composer self-update` does, but for simplicity and not getting hung up on these things, I@d prefer to have this as a future improvement IMO.

Reply:
Removed